controllers/event_watcher_registry.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package controllers
+
+import (
+	"context"
+
+	gocache "github.com/patrickmn/go-cache"
```

Comment:
I'm not familiar with this module. Do you think we should adopt it to replace some of the other internal caching implementations?

Reply:
It might be something to consider? I grabbed it since it's used a few other places in HashiCorp, and in [Vault in particular](https://github.com/search?q=org%3Ahashicorp+github.com%2Fpatrickmn%2Fgo-cache+repo%3Ahashicorp%2Fvault&type=code).